tasks/insertionTasks.ts

```diff
@@ -0,0 +1,260 @@
+        // Step 5: Generate PR list
+        const prList = await generatePRList(currentSHA, options.roslynEndSHA, options.roslynRepoPath, options);
+        console.log('PR List generated:', prList);
+
+        // Check if PR list is null or empty (generation failed or no matching PRs)
+        if (!prList) {
+            console.log('No PRs with required labels found or PR list generation failed. Skipping insertion.');
+            logWarning('No PRs with VSCode label found between the commits or PR list generation failed. Skipping insertion.');
+            return;
+        }
+
+        // Step 6: Update files
+        await updatePackageJson(options.roslynVersion);
+
+        // Step 7: Create branch and PR
+        const prTitle = `Bump Roslyn to ${options.roslynVersion} (${options.roslynEndSHA?.substring(0, 8)})`;
+
+        // Include build information in the PR description
+        let prBody = `This PR updates Roslyn to version ${options.roslynVersion} (${options.roslynEndSHA}).\n\n`;
+
+        // Add build link if build information is available
+        if (options.roslynBuildNumber && options.roslynBuildId) {
+            prBody += `Build: [#${options.roslynBuildNumber}](https://dev.azure.com/dnceng/internal/_build/results?buildId=${options.roslynBuildId})\n\n`;
+        }
+
+        // Add PR list
+        prBody += prList;
+
+        const commitMessage = `Bump Roslyn to ${options.roslynVersion} (${options.roslynEndSHA?.substring(0, 8)})`;
+
+        // Create the PR
+        const prNumber = await createBranchAndPR({
+            ...options,
+            commitSha: options.roslynEndSHA!,
+            targetRemoteRepo: 'vscode-csharp',
+            baseBranch: options.targetBranch || 'main',
+            newBranchName: `insertion/${options.roslynEndSHA}`,
+            githubPAT: options.githubPAT!,
+            dryRun: options.dryRun,
+            userName: options.userName,
+            email: options.email
+        }, prTitle, commitMessage, prBody);
+
+        // If PR was created and we're not in dry run mode, update the changelog with the PR number
+        if (prNumber && !options.dryRun) {
+            console.log(`PR #${prNumber} created. Updating changelog with PR link...`);
+
+            // Update changelog with PR number (single call)
+            await updateChangelog(options.roslynVersion, prList, prNumber);
+
+            // Create a second commit to include the updated changelog and push to update the PR
+            await git(['add', 'CHANGELOG.md']);
+            await git(['commit', '-m', `Update changelog with PR #${prNumber}`]);
+            await git(['push', 'target', `insertion/${options.roslynEndSHA}`]);
+
+            console.log(`Changelog updated with PR #${prNumber} link.`);
+        }
+
+    } catch (error) {
+        const errorMessage = error instanceof Error ? error.message : String(error);
+        logError(`Insertion failed: ${errorMessage}`);
+        if (error instanceof Error && error.stack) {
```

Comment:
Consider: Can printing the error stack trace be part of the logError function above?